lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -524,12 +524,13 @@ SmallVector<Value> getMultiDimOffset(Attribute layout, Location loc,
     return multiDimOffset;
   }
   if (auto mfmaLayout = layout.dyn_cast<AMDMfmaEncodingAttr>()) {
+    // TODO: extend to support dot3d
     auto multiDimBase =
         emitBaseIndexForLayout(loc, rewriter, layout, type, false);
     SmallVector<SmallVector<unsigned>> offsets;
     assert(rank == 2);
     SmallVector<Value> multiDimOffset(rank);
-    emitMfmaOffsetForCTA(mfmaLayout, offsets, multiDimCTAInRepId[0],
+    emitMfmaOffsetForCTA(mfmaLayout, offsets, 0, multiDimCTAInRepId[0],
```

Comment:
is there a way to not hardcode this ?

Reply:
This part of code is used by `lowerDistributedToDistributed`, which I feel is not used at all.
-  One use case is in the epilogue when we need to convert mfma to blocked before tt.store. The optimize_epilogue pass actually removes the conversion and uses mfma layout to do tt.store
- For other distributed layouts, like blocked->dotOp or mfma->dotOp, they should either be decomposed or a shortcut.

Do you have a use case for this conversion?

Reply:
Well, at least for `mma`, some ops do not support `mma` so if we have a `transpose` or `scan` on the result of a `dot`, we would have to convert mma to blocked. Might be a similar case for `mfma`. I would keep support.

Reply:
This is interesting. 
If this conversion happens inside a loop, the traffic to and from shared memory will harm perf a lot. So I guess in this case, you'd rather pay the price for perf than supporting mma for those ops.
I can support it anyway, but do you have a test so that I can verify the results?

Reply:
Not sure if we have a test, maybe @ThomasRaoux knows.

Reply:
I believe we should be tested in `test_convert2d` as we are testing different combination of convert including mma layouts. If this is not tested we should definitely add a case there.

> If this conversion happens inside a loop, the traffic to and from shared memory will harm perf a lot. So I guess in this case, you'd rather pay the price for perf than supporting mma for those ops.

well there is always different ways to propagate and reduce the cost but we want functionality first so we should support this no matter what. When we run into performance problems we will address those.

Reply:
Fair enough. I get the point that we should support conversions between distributed layout. Therefore, we should keep this file.
> I believe we should be tested in test_convert2d as we are testing different combination of convert including mma layouts. If this is not tested we should definitely add a case there.

I'd suggest we do it in future PRs, since this PR is about dot3d. WDYT?